sdktests/server_side_eval_all_flags.go

```diff
@@ -292,7 +295,7 @@ func doServerSideAllFlagsDetailsOnlyForTrackedFlagsTest(t *ldtest.T) {
 		},
 		"$valid": true
 	}`
-	assert.JSONEq(t, expectedJSON, string(resultJSON))
+	m.In(t).Assert(resultJSON, m.JSONStrEqual(expectedJSON))
```

Comment:
I switched to using the `matchers` version of this assertion because I like the output better. For instance, if we see a property we didn't expect, `assert.JSONEq` will print something like this:

```
  Not equal:
  expected: map[string]interface {}{"$flagsState":map[string]interface {}{"flag1":map[string]interface {}{"variation":1}, "flag2":map[string]interface {}{"reason":map[string]interface {}{"kind":"OFF"}, "trackEvents":true, "variation":2, "version":200}, "flag3":map[string]interface {}{"debugEventsUntilDate":1.6433153693e+12, "reason":map[string]interface {}{"kind":"OFF"}, "variation":3, "version":300}}, "$valid":true, "flag1":"value1", "flag2":"value2", "flag3":"value3"}
  actual  : map[string]interface {}{"$flagsState":map[string]interface {}{"flag1":map[string]interface {}{"variation":1, "version":100}, "flag2":map[string]interface {}{"reason":map[string]interface {}{"kind":"OFF"}, "trackEvents":true, "variation":2, "version":200}, "flag3":map[string]interface {}{"debugEventsUntilDate":1.6433153693e+12, "reason":map[string]interface {}{"kind":"OFF"}, "variation":3, "version":300}}, "$valid":true, "flag1":"value1", "flag2":"value2", "flag3":"value3"}
  Diff:
  --- Expected
  +++ Actual
  @@ -2,4 +2,5 @@
  (string) (len=11) "$flagsState": (map[string]interface {}) (len=3) {
  -  (string) (len=5) "flag1": (map[string]interface {}) (len=1) {
  -   (string) (len=9) "variation": (float64) 1
  +  (string) (len=5) "flag1": (map[string]interface {}) (len=2) {
  +   (string) (len=9) "variation": (float64) 1,
  +   (string) (len=7) "version": (float64) 100
  },
```

But with `matchers.JSONStrEqual` you get this:
```
  JSON values at "$flagsState"."flag1"."version": expected = <absent>, actual = 100
  full value was: {"$flagsState":{"flag1":{"variation":1,"version":100},"flag2":{"reason":{"kind":"OFF"},"trackEvents":true,"variation":2,"version":200},"flag3":{"debugEventsUntilDate":1643315408682,"reason":{"kind":"OFF"},"variation":3,"version":300}},"$valid":true,"flag1":"value1","flag2":"value2","flag3":"value3"}
```